src/compilation/scopeData.ts

```diff
@@ -1,19 +1,17 @@
-import { TemplateContent, TemplateData } from '../templateData';
-import { last } from '../utils';
+import { TemplateContent, TemplateData } from "../templateData";
+import { last } from "../utils";
 
-const getProp = require('lodash.get');
+const getProp = require("lodash.get");
 
 export class ScopeData {
-    
     public readonly path: (string | number)[] = [];
     public readonly allData: TemplateData;
 
     constructor(data: TemplateData) {
         this.allData = data;
     }
 
-    public getScopeData(): TemplateContent | TemplateData[] {
-
+    public getScopeData<T = TemplateContent | TemplateData[]>(): T{
```

Comment:
I think this should be "T extends TemplateContent | TemplateData[]" with no default value or 'unknown' as the deafult